psh/tools/psh.py

```diff
@@ -195,15 +195,24 @@ def ls(pexpect_proc, dir=''):
         # temporary solution to match line with missing `---` in owner position (issue #254)
         line = line.replace('    ---', '--- ---')
         try:
-            permissions, _, owner, _, _, month, mday, time, name = line.split(maxsplit=9)
+            permissions, _, owner, _, _, month, mday, time_year, name = line.split(maxsplit=9)
         except ValueError:
             assert False, f'wrong ls output: {line}'
 
         # Name is printed with ascii escape characters - remove them
         name = re.sub(CONTROL_CODE, '', name)
 
-        # the year is set to default 1970, 1 min resolution
-        file_datetime = datetime.strptime(f'1970 {month} {mday} {time}:00', '%Y %b %d %X')
+        # time_year can contain either time (HH:MM) or year (YYYY)
+        # If year was not printed, replace it with default
+        if ':' in time_year:
+            time = time_year
+            year = default_year
+        else:
+            time = '00:00'
```

Comment:
Hmm, that's not 00:00 in fact, but we could leave it with a proper comment showing that the exact value does not matter when we are comparing two dates without a specific HH:MM

Reply:
I solved this by passing the timestamp resolution along with the timestamp, so it can be compared correctly by `assert_mtime`.